feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/login/SignupActivity.java

```diff
@@ -1,33 +1,83 @@
-package com.example.feelslikemonday.ui.login;
-
-import androidx.appcompat.app.AppCompatActivity;
-
-import android.content.Intent;
-import android.os.Bundle;
-import android.view.View;
-
-import com.example.feelslikemonday.MainActivity;
-import com.example.feelslikemonday.R;
-
-public class SignupActivity extends AppCompatActivity {
-
-    public void cancelSignup(View view) {
-        finish();
-
-    }
-
-    public void confirmSignup(View view) {
-        Intent myIntent = new Intent(this, MainActivity.class);
-        this.startActivity(myIntent);
-    }
-
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_signup);
-
-    }
-
-
-}
+package com.example.feelslikemonday.ui.login;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+
+import com.example.feelslikemonday.DAO.FollowPermissionDAO;
+import com.example.feelslikemonday.DAO.FollowRequestDAO;
+import com.example.feelslikemonday.DAO.UserDAO;
+import com.example.feelslikemonday.DAO.VoidCallback;
+import com.example.feelslikemonday.MainActivity;
+import com.example.feelslikemonday.R;
+import com.example.feelslikemonday.model.FollowPermission;
+import com.example.feelslikemonday.model.FollowRequest;
+import com.example.feelslikemonday.model.User;
+
+/*User Preferences designed according to tutorial on:
+https://www.journaldev.com/9412/android-shared-preferences-example-tutorial
+ */
+
+public class SignupActivity extends AppCompatActivity {
+
+    EditText signupUsername;
+    EditText signupPassword;
+    String username;
+    SharedPreferences pref;
+    public static final String PREFS_NAME = "user_preferences";
+    public static final String USERNAME_KEY = "username_key";
+
+    public void cancelSignup(View view) {
+        finish();
+    }
+
+    public void confirmSignup(View view) {
+        // THIS DOES NOT CHECK FOR UNIQUE USERNAMES YET. SHOULD CHECKING GO INTO SERVICE?
```

Comment:
Checking should go into service

Reply:
I did not write this, It's from when I pulled from davinas branch yesterday. Literally the only stuff I added is the stuff that relates to shared preference.

Reply:
I prefer not to change her code because she said she's working on login now.

Reply:
I saw it as green and assumed that the code was added in. Disregard comment